File: backend/src/main/java/es/udc/tfgproject/backend/model/services/ProductCatalogService.java
package es.udc.tfgproject.backend.model.services;

import java.util.List;

import es.udc.tfgproject.backend.model.entities.Product;
import es.udc.tfgproject.backend.model.entities.ProductCategory;

public interface ProductCatalogService {

	List<Product> findProducts(Long companyId, Long productCategoryId, String keywords);

	List<ProductCategory> findCompanyProductCategories(Long companyId);
}

File: frontend/src/backend/productCatalogService.js
import {config, appFetch} from './appFetch';

export const findProducts = (companyId, {productCategoryId, keywords}, 
    onSuccess) => {

    let path = `/productCatalog/products?companyId=${companyId}`;

    path += productCategoryId ? `&productCategoryId=${productCategoryId}` : "";
    path += keywords.length > 0 ? `&keywords=${keywords}` : "";

    appFetch(path, config('GET'), onSuccess);

}

export const findCompanyProductCategories = (companyId, onSuccess) => 
    appFetch(`/productCatalog/products/categories?companyId=${companyId}`, 
        config('GET'), onSuccess);
